lib/models/Virtualenv.js

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+var childProcess = require('child_process');
+var async = require('async');
+var notif = require('./Notifcations');
+var path = require('path');
+var logger = require('log4js').getLogger('CloudifyCLI');
+
+var Virtualenv = function() {
+
+    /**
+     * Initialize virtualenv
+     * * *
+     * @param {object} options - cmmander options
+     * @param {function} fnCallback
+     */
+    function init(options, fnCallback) {
+        if(options.env) {
+            async.waterfall([
+
+                function(callback) {
+                    notif.info('Initialize virtual environment...')
+                    initEnvironment(options.prefix, function(err){
+                        if(err) {
+                            notif.faild('Initialize failed.');
+                        }
+                        else {
+                            notif.ok('Initialize success.');
+                            callback(null);
+                        }
+                    });
+                }
+
+            ], fnCallback);
+        }
+        else {
+            fnCallback(null);
+        }
+    }
+
+    /**
+     * Initialize environment
+     * * * 
+     * @param {string} env - environment name
+     * @param {function} callback
+     */
+    function initEnvironment(env, callback) {
+        var exec = childProcess.spawn('virtualenv', [env]);
+
+        exec.stdout.on('data', function (data) {
+            logger.trace('stdout: ' + data);
+        });
+
+        exec.stderr.on('data', function (data) {
+            logger.trace('stderr: ' + data);
+        });
+
+        exec.on('close', function (code) {
+            callback(code !== 0 ? true : null);
+            exec.stdin.end();
+        });
+    }
+    
+    function spawn(cmd, options, args, callback) {
+        var stdout;
+        var stderr;
+        
+        if(typeof(args) === 'function') {
+            callback = args;
+            args = {};
+        }
+
+        switch(options.env) {
+            case true:
+                var exec = childProcess.spawn('bash', ['-c', 'source ' + path.resolve(options.prefix) + '/bin/activate; ' + cmd], args || {});
```

Comment:
need to check if prefix is defined. 
true that we fail in command parser if it does not, but every function should revalidate.